### PR TITLE
authz: refactor the pilot authz plugin

### PR DESCRIPTION
**Pull request by yangminzhu**
#22060 

This PR refactor the pilot authz plugin:
- simplify the builder API as we only need to support the beta API
- reorganize the tests to compare the entire filter config and also support auto refresh golden files
- the golden files are only changed to include the full RBAC filter config output, no semantic change in this PR. Check the 2nd commit: https://github.com/istio/istio/pull/22537/commits/9195f412547b3f89cb3d1bd28e23f5f14d705652

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/security/authz/builder/builder.go [hunk 1]
```diff
@@ -15,113 +15,141 @@
...
 	return filters
 }
 
-// nolint: interfacer
-func createHTTPFilter(config *envoyRbacHttpPb.RBAC) *httpFilterPb.HttpFilter {
-	if config == nil {
-		return nil
-	}
+// BuildTCP returns the RBAC TCP filters built from the authorization policy.
+func (b Builder) BuildTCP() []*tcppb.Filter {
```
Comment (diemtvu) [inline pilot/pkg/security/authz/builder/builder.go]: [nit] I actually like the old name better as it's more descriptive.

File: pilot/pkg/security/trustdomain/bundle.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 )
 
 var (
-	rbacLog = istiolog.RegisterScope("rbac", "rbac debugging", 0)
+	authzLog = istiolog.RegisterScope("authorization", "Istio Authorization Policy", 0)
```
Comment (diemtvu) [inline pilot/pkg/security/trustdomain/bundle.go]: Should this be a separate logger by itself?

Comment (yangminzhu) [reply pilot/pkg/security/trustdomain/bundle.go]: yeah, probably it should be a different logger as it's under the trustdomain pkg, or maybe the better way is to remove it from the pkg directory and let the caller to log, either way it's better to be done in a separate PR.

File: pilot/pkg/security/authz/builder/builder.go (hunk 1 above)
Comment (yangminzhu) [reply pilot/pkg/security/authz/builder/builder.go]: You mean `BuildTCPFilters()`? Hmm, we can change that later.

**Pull request merged**